src/routes/popup/agents/components/ao-yield/DateSelectorModal.tsx

```diff
@@ -0,0 +1,792 @@
+    const remainingCells = totalCells - (firstDay + daysInMonth);
+
+    for (let day = 1; day <= remainingCells; day++) {
+      const isNextMonthSelected = isDateSelectedOtherMonth(day, 1);
+      const isNextMonthStart = isStartDateOtherMonth(day, 1);
+      const isNextMonthEnd = isEndDateOtherMonth(day, 1);
+      const isNextMonthWeekStart = isStartOfWeekInRangeOtherMonth(day, 1);
+      const isNextMonthWeekEnd = isEndOfWeekInRangeOtherMonth(day, 1);
+
+      // Check if this next month date is disabled (before today)
+      const nextMonthDate = new Date(currentDate.getFullYear(), currentDate.getMonth() + 1, day);
+
+      // Check if this next month date is in range
+      const isNextMonthInRange =
+        startDate && endDate && nextMonthDate >= startDate && nextMonthDate <= endDate && !isNextMonthSelected;
+      const today = new Date();
+      today.setHours(0, 0, 0, 0);
+      const isNextMonthDisabled = nextMonthDate < today;
+
+      days.push(
+        <CalendarDay
+          key={`next-${day}`}
+          $isOtherMonth
+          $isSelected={isNextMonthSelected}
+          $isInRange={isNextMonthInRange}
+          $isStart={isNextMonthStart}
+          $isEnd={isNextMonthEnd}
+          $isWeekStart={isNextMonthWeekStart}
+          $isWeekEnd={isNextMonthWeekEnd}
+          $connectToPrev={false}
+          $connectToNext={false}
+          $isSingleDate={
+            (isNextMonthStart && isNextMonthEnd) || (isNextMonthStart && !endDate) || (isNextMonthEnd && !startDate)
+          }
+          $isDisabled={isNextMonthDisabled}
+          onClick={() => !isNextMonthDisabled && handleOtherMonthDateClick(day, 1)}>
+          <Text
+            size="sm"
+            style={{
+              color: isNextMonthSelected ? "#fff" : isNextMonthDisabled ? theme.tertiaryText : theme.secondaryText,
+            }}
+            weight={isNextMonthSelected ? "semibold" : "medium"}
+            noMargin>
+            {day}
+          </Text>
+        </CalendarDay>,
+      );
+    }
+
+    // Add empty cells to ensure we always have 6 rows (42 cells total)
+    const totalDays = days.length;
+    const emptyCellsNeeded = 42 - totalDays;
+
+    for (let i = 0; i < emptyCellsNeeded; i++) {
+      days.push(<EmptyCalendarDay key={`empty-${i}`} />);
+    }
+
+    return days;
+  };
+
+  return (
+    <Container>
+      <CalendarContainer>
+        <MonthNavigation>
+          <NavButton
+            onClick={() => navigateMonth(-1)}
+            $disabled={
+              currentDate.getFullYear() <= new Date().getFullYear() && currentDate.getMonth() <= new Date().getMonth()
+            }>
+            <ChevronLeft width={20} height={20} color={theme.primaryText} />
+          </NavButton>
+          <Text size="lg" weight="semibold" noMargin>
+            {monthNames[currentDate.getMonth()]} {currentDate.getFullYear()}
+          </Text>
+          <NavButton onClick={() => navigateMonth(1)}>
```

Comment:
Similarly to my comment in the range input, it would be good to be able to hold down these buttons to quickly move back/forward, and/or to be able to change the month/year manually (assuming going further into the future is a common scenario).